examples/src/main/java/com/mapbox/navigation/examples/NavigationApplication.kt

```diff
@@ -61,6 +58,5 @@ class NavigationApplication : MultiDexApplication() {
         }
 
         Mapbox.getInstance(applicationContext, mapboxAccessToken)
-        MapboxNativeNavigatorImpl.create(DeviceProfile.Builder().build(), NavigatorConfig(null), MapboxLogger)
```

Comment:
This seems unnecessary. @kmadsen do you remember why this was added in the first place? Is it safe to remove?

Reply:
Yeah it resolves a crash in the examples. The first update location before an example is opened

https://github.com/mapbox/mapbox-navigation-android/pull/3071
```
E/AndroidRuntime: FATAL EXCEPTION: pool-1-thread-2
Process: com.mapbox.navigation.examples, PID: 32302
kotlin.KotlinNullPointerException
at com.mapbox.navigation.navigator.internal.MapboxNativeNavigatorImpl.updateLocation(MapboxNativeNavigatorImpl.kt:79)
at com.mapbox.navigation.core.internal.trip.session.MapboxTripSession$updateRawLocation$2.invokeSuspend(MapboxTripSession.kt:414)
at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:56)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
at java.lang.Thread.run(Thread.java:919)
```

Reply:
That was added before we created the MapboxNavigationProvider.

There are other problems that would benefit from a public navigator reset function, coming from nav native. https://github.com/mapbox/mapbox-navigation-native/issues/1061

That issue is stale so we may need to push for it

Reply:
Try toggling history recording? 
https://github.com/mapbox/mapbox-navigation-android/blob/55793266db129c701f9b09963a629e6b9b72216d/examples/src/main/java/com/mapbox/navigation/examples/MainActivity.kt#L107

Reply:
Here's the reproduce steps
https://github.com/mapbox/mapbox-navigation-android/issues/3068

Reply:
Fixed using `MapboxNavigationProvider.create` instead of the `internal` `MapboxNativeNavigatorImpl.create`. BTW this is an "issue" of how the `examples` app is implemented that we should revisit as part of the `examples` test app refactor cc @abhishek1508